files/scripts/write_standby.py

```diff
@@ -91,6 +92,21 @@ def is_dualtor(self):
 
         return 'subtype' in metadata and 'dualtor' in metadata['subtype'].lower()
 
+    @property
+    def is_mocked_dualtor(self):
+        """
+        Checks if script is running on a mocked dual ToR system
+        """
+        if not self.is_dualtor:
+            return False
+        subtype = ''
+        with open("/etc/sonic/config_db.json", 'r') as config_file:
+            file_data = json.load(config_file)
+            if 'subtype' in file_data['DEVICE_METADATA']['localhost']:
+                subtype = file_data['DEVICE_METADATA']['localhost']['subtype']
+
+        return 'dualtor' not in subtype.lower()
```

Comment:
i'm not sure what's the difference between `is_dualtor `and `is_mocked_dualtor`. Both are checking for `subtype`.  Also, why opening `config_db.json` file? This may or may not be the actual running config.

Reply:
is_dualtor checks the config database that's the running config, and is_mocked_dualtor checks the saved start config config_db.json, 
for the mocked case, at the timing of applying mock, it justs change the running config but doesn't save it.

here are my dumped data logs in two functions, we can see, is_dualtor has dualtor data like 'subtype': 'DualToR' and 'peer_switch': 'switch_hostname', while the is_mocked_dualtor doesn't have them.

In is_dualtor check:
meta {'bgp_asn': '64601', 'buffer_model': 'traditional', 'cloudtype': 'None', 'default_bgp_status': 'up', 'default_pfcwd_status': 'disable', 'deployment_id': '1', 'docker_routing_config_mode': 'separated', 'hostname': 'mth-t0-64', 'hwsku': 'Cisco-8102-C64', 'mac': 'BC:E7:12:82:03:90', 'peer_switch': 'switch_hostname', 'platform': 'x86_64-8102_64h_o-r0', 'region': 'None', 'subtype': 'DualToR', 'synchronous_mode': 'enable', 'type': 'ToRRouter'}

in is_mocked_dualtor check:
local {'bgp_asn': '64601', 'buffer_model': 'traditional', 'cloudtype': 'None', 'default_bgp_status': 'up', 'default_pfcwd_status': 'disable', 'deployment_id': '1', 'docker_routing_config_mode': 'separated', 'hostname': 'mth-t0-64', 'hwsku': 'Cisco-8102-C64', 'mac': 'BC:E7:12:82:03:90', 'platform': 'x86_64-8102_64h_o-r0', 'region': 'None', 'synchronous_mode': 'enable', 'type': 'ToRRouter'}

Reply:
thanks for explaining. @theasianpianist , do you think if there is a different approach for this here? I'm concerned about running this in real production flow where for some reason, saved file doesn't have this attribute whereas running config has this.

Reply:
Hi @theasianpianist, do you have a chance to help review it? thanks

Reply:
I agree with Prince here, plus I don't think it makes sense to change the image for a mock dualtor test issue. Do you see this problem on true dualtor testbeds as well?

Reply:
The problem doesn't happen on true dualtor testbed, and only the mock case (simulating dualtor on T0 testbed) has this problem.
For the case of "saved file doesn't have this attribute whereas running config has this", it's the way how the mocking/simulating works currently, that should be by the infra design. so the real issue is, the simulating doesn't work perfectly and then hit this problem at the SWSS restarting flow, 
this solution here is just to simply skip the check that's not necessary for the mocking case so the cases can pass.
the code change here is just to check the "dualtor" subtype, for the subtype attribute, i think only the mocking case will change it, at any other case it may be changed in running time on real production?

Reply:
how is it working currently? mocked dualtor has been running for quite a while now.

Reply:
it's an issue exposed recently by some other latest change. there was no problem before that change.
There was a change for wait_for_tunnel() about a couple of months ago, the timeout in wait_for_tunnel() was changed from 60s to 90s, the change itself doesn’t have any problem, but it exposes this issue, when the timeout was 60s, even there was no tunnel ready and the function wait_for_tunnel()  got timeout, the stopping bgp service would not get timeout so then swss could restart all services normally, and related cases could pass.
now the timeout is changed to 90s, the issue gets exposed.

Reply:
Hi @prsunny @theasianpianist , any suggestion for the solution? thanks

Reply:
@prsunny  @theasianpianist  - could you please check the latest comments from @yucgu  and provide us the next steps?

Reply:
@prsunny  @theasianpianist  - could you please help in approving this?